tests/js/plugin/enhancedanchoring-test.coffee

```diff
@@ -0,0 +1,198 @@
+assert = chai.assert
+
+# In order to be able to create highlights,
+# the Annotator.TextHighlight class must exist.
+# This class is registered then the TextHighlights plugin
+# is initialized, so we will do that.
+th = new Annotator.Plugin.TextHighlights()
+th.pluginInit()
+
+class FakeAnchor extends Annotator.Anchor
+
+  _getSegment: -> "html segment for " + @id
+
+  constructor: (manager, annotation, target) ->
+    super manager, annotation, target, 42, 42,
+      "fake quote for" + target.id
+
+    @id = "fake anchor for " + target.id
+
+describe 'Annotator.Plugin.EnhancedAnchoring', ->
+  sandbox = null
+
+  createAnchoringManager = ->
+    annotator =
+      publish: (event) ->
+
+    am = new Annotator.Plugin.EnhancedAnchoring()
+    am.annotator = annotator
+    am.pluginInit()
+
+    am._chooseAccessPolicy()
```

Comment:
> > +    am._chooseAccessPolicy()
>
> What's the public method that triggers this call? I had a look and
> it's called by |_scan| but I couldn't see what calls scan.
>

There is no public method that triggers this call. scan is called by
guest.coffee and host.coffee.

The separation of private and public methods (and responsibilities of
d-t-m, anchoring manager and h)
are confusing here.

  *  * *

This is yet again something that has been cleaned up on our newer
Annotator branches (observer and friends),
but I never did it on the current legacy branch, because I was always
laboring under the assumption that we
are going to move to those other branches soon.

Maybe it's time to start to look over those changes and fetch the
improvements that are independent of the actual d-t-m changes.

  * * *

For now, I'll just make _chooseAccessPolicy() public, because it's legal
and harmless to call it from the outside.